sycl/test-e2e/lit.cfg.py

```diff
@@ -308,6 +308,33 @@ def open_check_file(file_name):
 if sp[0] == 0:
     config.available_features.add("preview-breaking-changes-supported")
 
+# Check if clang is built with ZSTD and compression support.
```

Comment:
There is a way simpler way. Use [`lit.site.cfg.py.in`](https://github.com/intel/llvm/blob/sycl/sycl/test-e2e/lit.site.cfg.py.in) to propagate a value of CMake variable into this python script.

I would also explore how LLVM propagates that. There are tests in LLVM which require `zstd` feature, so I wonder if we can call some LIT helper to get this feature automatically propagated into LIT for us

Reply:
As discussed offline, passing `LLVM_ENABLE_ZSTD` from CMake to LIT won't work here because E2E tests can be built standalone, like what we do in CI.
LLVM seems to pass CMake Variables to LIT: https://github.com/llvm/llvm-project/blob/main/llvm/test/lit.site.cfg.py.in#L37

Reply:
Yeah, good point. I still wonder if there is a simpler way (i.e. some existing helper for running an executable and getting its output, but if no, then we will have to leave with what we have.

BTW, I'm sure that compiler is able to read the program from `stdin`, so you can maybe save on file operations here